app/helpers/application_html_formatters_helper.rb

```diff
@@ -74,7 +75,9 @@ def format_code_block(code, language = nil)
       end
     end
 
-    format_with_pipeline(DefaultCodePipeline, code)
+    formatted_code = DefaultCodePipeline.to_document(code)
```

Comment:
so in this case, the thing to do is to change all pipelines from being class constants to being class methods. Create a new pipeline every time one needs to be created, then.